src/log/log_functions.c

```diff
@@ -1483,6 +1490,8 @@ static void flush_log_file(int i, bool periodic, bool close_fd)
 		return;
 	}
 
+  rotate_log(fd, file); /* Rotate log files */
+
 	if (periodic) {
```

Comment:
If rotate has closed the fd then the following calls are no longer needed.

Reply:
updating close_fd flag to false if rotate closing fd.